js/ux-capture.js

```diff
@@ -22,91 +33,114 @@ window.UX = (function() {
+          if (isUserTimingSupported) {
+            // record a measure using W3C User Timing API
+            window.performance.measure(
+              zone.label,
+              "navigationStart",
+              zone.lastMarkLabel
+            );
+
+            // if callback is specified, call it with zone label
+            if (onMeasure) {
+              onMeasure(zone.label);
+            }
+          }
+        });
 
-                  // remember last mark that was recorded within a zone, overriding previous one
-                  zone.lastMarkLabel = mark.label;
+        return zone;
+      }
+    });
+  };
 
-                  resolve(mark.label);
-                };
-              })
-          );
+  window.UX.mark = function(markLabel) {
+    const mark = getMark(markLabel);
 
-          // only if all marks were recorded (and promises resolved), go ahead and record the measure ending with last recorded mark
-          Promise.all(promises).then(() => {
-            if (isUserTimingSupported) {
-              // record a measure using W3C User Timing API
-              performance.measure(
-                zone.label,
-                "navigationStart",
-                zone.lastMarkLabel
-              );
-
-              // if callback is specified, call it with zone label
-              if (onMeasure) {
-                onMeasure(zone.label);
-              }
-            }
-          });
-
-          return zone;
-        }
-      });
-    },
-    mark: function(markLabel) {
-      const mark = getMark(markLabel);
+    if (mark) {
+      mark.record();
+    }
+  };
 
-      if (mark) {
-        mark.record();
-      }
-    },
-    config: function(configuration) {
-      if (!configuration || typeof configuration !== "object") {
-        return;
-      }
+  window.UX.config = function(configuration) {
+    if (!configuration || typeof configuration !== "object") {
+      return;
+    }
 
-      if (typeof configuration.onMark === "function") {
-        onMark = configuration.onMark;
-      }
+    if (typeof configuration.onMark === "function") {
+      onMark = configuration.onMark;
+    }
 
-      if (typeof configuration.onMeasure === "function") {
-        onMeasure = configuration.onMeasure;
-      }
+    if (typeof configuration.onMeasure === "function") {
+      onMeasure = configuration.onMeasure;
     }
   };
-})();
+
+  //
+  // Export UX Capture to the appropriate location.
+  //
+  // When included directly via a script tag in the browser, we're good as we've been
+  // updating the window.UX object.
+  //
+  if (typeof define === "function" && define.amd) {
+    //
+    // AMD / RequireJS
+    //
+    define([], function() {
+      return window.UX;
+    });
+  } else if (
+    typeof module !== "undefined" &&
+    typeof module.exports !== "undefined"
+  ) {
+    //
+    // Node.js
+    //
+    module.exports = window.UX;
+  }
+})(typeof window !== "undefined" ? window : undefined);
```

Comment:
Not sure what this ternary is doing - if `window` is `undefined`, then `undefined` gets passed in, which means you could just pass in `window` either way?

Reply:
Yeah, that got me too - if `typeof window` is undefined (e.g. it was never declared), you can't pass it in so you have to pass in `undefined` instead, tried without it, get a warning.